### PR TITLE
ci: move blob storage to azure for builds

### PR DESCRIPTION
**Pull request by rsmitty**
This PR moves blob storage to azure.

### CODE REVIEW

File: .drone.jsonnet [hunk 1]
```diff
@@ -267,16 +267,15 @@ local push_latest = {
 
 local save_artifacts = {
   name: 'save-artifacts',
-  image: 'autonomy/build-container:latest',
+  image: 'mcr.microsoft.com/azure-cli:latest',
```
Comment (frezbo) [inline .drone.jsonnet:270]: doesn't pur image already has azure-cli?

Comment (rsmitty) [reply .drone.jsonnet]: yeah it does. just missed it when I went looking earlier. thx.

**Comment by rsmitty** [comment]
needs https://github.com/siderolabs/extensions/pull/128 merged first.

**Review by frezbo — APPROVED** [review]

**Pull request merged**